components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngine.kt

```diff
@@ -158,6 +160,11 @@ class GeckoEngine(
         this.webExtensionDelegate = webExtensionDelegate
 
         val tabsDelegate = object : WebExtensionController.TabDelegate {
+            // We use this map to:
+            // - Ensures a web extension can only close tabs it has opened
+            // - Find the engine session of a given gecko session
```

Comment:
Can't we also do this lookup by going over the engine sessions of all tabs (with a cast)?

Reply:
OK, as discussed, since we can't currently perform this lookup in an acceptable way, we will keep this map here and live with the limitation that a web extension can only close tabs it has opened. This problem will go away once the browser-state migration is done and the engine has access to the browser store. The issue is tracked here: https://github.com/mozilla-mobile/android-components/issues/4965